src/three-components/SmoothControls.ts

```diff
@@ -154,27 +154,30 @@ export class Damper {
   update(
       x: number, xGoal: number, timeStepMilliseconds: number,
       xNormalization: number): number {
-    if (timeStepMilliseconds >= DECAY_MILLISECONDS) {
-      // This clamps at the point where a large time step would cause the
-      // discrete step to overshoot.
-      this[$velocity] = 0;
+    if (!x) {
       return xGoal;
     }
     if (timeStepMilliseconds < 0) {
       return x;
     }
-    // Critically damped
-    const acceleration = NATURAL_FREQUENCY * NATURAL_FREQUENCY * (xGoal - x) -
-        2 * NATURAL_FREQUENCY * this[$velocity];
-    this[$velocity] += acceleration * timeStepMilliseconds;
-    if (Math.abs(this[$velocity]) < NIL_SPEED * xNormalization &&
-        acceleration * (xGoal - x) <= 0) {
+    // Exact solution to a critically damped second-order system, where:
+    // acceleration = NATURAL_FREQUENCY * NATURAL_FREQUENCY * (xGoal - x) -
+    // 2 * NATURAL_FREQUENCY * this[$velocity];
+    const c1 = (x - xGoal);
+    const c2 = this[$velocity] + NATURAL_FREQUENCY * c1;
+    const c3 = c1 + timeStepMilliseconds * c2;
```

Comment:
[Please use descriptive names for these terms](https://github.com/GoogleWebComponents/model-viewer/wiki/Code-conventions#naming)

Reply:
These don't have names, they are only saved here so we don't have to do the calculations multiple times. I can call c1 deltaX, but c2 and c3 are physically meaningless as far as I can tell.

Reply:
I renamed them so that at least you can tell their units.